controller/backup_controller.go

```diff
@@ -116,6 +121,50 @@ func (bc *BackupController) enqueueBackupForMonitor(key string) {
 	bc.queue.Add(key)
 }
 
+func (bc *BackupController) enqueueInstanceManagerChange(obj interface{}) {
+	im, isInstanceManager := obj.(*longhorn.InstanceManager)
+	if !isInstanceManager {
+		deletedState, ok := obj.(cache.DeletedFinalStateUnknown)
+		if !ok {
+			utilruntime.HandleError(fmt.Errorf("received unexpected obj: %#v", obj))
+			return
+		}
+
+		// use the last known state, to enqueue, dependent objects
+		im, ok = deletedState.Obj.(*longhorn.InstanceManager)
+		if !ok {
+			utilruntime.HandleError(fmt.Errorf("DeletedFinalStateUnknown contained invalid object: %#v", deletedState.Obj))
+			return
+		}
+	}
+
+	imType, err := datastore.CheckInstanceManagerType(im)
+	if err != nil || imType != longhorn.InstanceManagerTypeEngine {
+		return
+	}
+
+	if im.Status.OwnerID != bc.controllerID {
+		return
+	}
+
+	if im.Status.CurrentState == longhorn.InstanceManagerStateStarting || im.Status.CurrentState == longhorn.InstanceManagerStateRunning {
+		return
+	}
+
+	backups, err := bc.ds.ListBackupsRO()
```

Comment:
This will be a potential concern about the cost if there are tons of backup resources needed to query. Is there a way to filter by owner instead?

Reply:
cc @shuo-wu @derekbit @PhanLe1010 will this be a concern?

Reply:
Seems we can only filter by label selector.
https://github.com/longhorn/longhorn-manager/blob/master/k8s/pkg/client/listers/longhorn/v1beta2/backup.go#L63

Reply:
It seems that the lister (cache) doesn't support the field selector.
Not sure if we need to add a label for the owner ID